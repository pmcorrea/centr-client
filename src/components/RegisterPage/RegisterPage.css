label {
	color: white;
}

.red {
	color: white;
}