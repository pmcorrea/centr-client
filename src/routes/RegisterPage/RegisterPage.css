/* label {
	color: #eee;
} */


.RegisterForm__form_contianer {
	flex-direction: column;
	width: 100vw;
  	align-items: center;
	  justify-content: center;
	  z-index: 100;
	  position: relative;
}

.RegisterForm {
	/* background-color: rgba(214, 214, 214, 0.856); */
	width: 400px;
	margin: 0 auto;
	width: 80vw;
	z-index: 100;
}

.input_field {
	border-top: 0px;
	border-right: 0px;
	border-left: 0px;
	border-bottom: 2px rgba(214, 214, 214, 0.63) solid;
	margin-top: 10px;
	color: black;
	/* background-color: rgb(255, 255, 255); */
	font-size: 20px;
	padding: 8px;
	width: 100%;
	border-radius: 5px;
	cursor: pointer;
	border: 2px solid rgb(112, 39, 161);
}

.input_field::placeholder {
	color: gray;
}

.register_button {
	width: 100%; 
	padding: 15px;
	font-size: 15px;
	background-color: rgb(112, 39, 161);
	color: white;
	border-radius: 5px;
	border: 0px solid white;
	margin-top: 10px; 
	font-weight: bold;
	cursor: pointer;
}

.register_button:disabled {
	background-color: rgb(147, 122, 163);
}

#file-upload_label {
	color: black;
	margin-top: 10px;
}

.registration__container {
	padding-top: 20px;
}

.registration__validation_box{
	padding: 20px 0px;
	display: flex;
	flex-direction: column;
	justify-content: flex-start;
	align-items: flex-start;
	background-color: white;
}

.validation_rules {
	margin: 0px;
	font-size: 15px;
	color: rgb(112, 39, 161);
}