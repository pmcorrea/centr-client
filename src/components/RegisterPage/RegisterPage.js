import React, {Component} from 'react'
import { Button, Input } from "../../Utils/Utils";
import './RegisterPage.css'
import MainContext from "../../contexts/MainContext";
import AuthApiService from "../../services/auth-api-service";


export default class RegisterPage extends Component {

	static contextType = MainContext

	state = { 
		error: null,
		user_name: null,
		password: null,
	};

	handleRegister(e) {
		e.preventDefault()
		this.setState({ error: null })
		const {user_name, password} = this.state
		console.log(user_name, password)

		AuthApiService.postRegister({
			user_name: user_name, 
			password: password,
		})
			.then(res => {
				// select folders and notes
				this.setState({user_name: null, password: null}, console.log(res))
			})
			.catch(res => {
				this.setState({ error: res.error });
			})
	}

	setUsername(username) {
		this.setState({
			user_name: username
		})
	}

	setPassword(password) {
		this.setState({
			password: password
		})
	}

	render() {
		const { error } = this.state;
		return (
			<>
				<form className="LoginForm" onSubmit={e => this.handleRegister(e)} id="form_register">
				<div role="alert">{error && <p className="red">{error}</p>}</div>

				<div className="user_name">
					<label htmlFor="LoginForm__user_name">User name</label>
					<Input required name="user_name" id="LoginForm__user_name" value={this.state.user_name} onChange={e => this.setUsername(e.target.value)}></Input>
				</div>

				<div className="password">
				<label htmlFor="LoginForm__password">Password</label>
				<Input
					required
					name="password"
					type="password"
					id="LoginForm__password"
					onChange={e => this.setPassword(e.target.value)}
					// value={this.state.password} 
				></Input>
				<Input
					required
					name="password_confirm"
					type="password"
					id="LoginForm__password_confirm"
					// value={this.state.password}
				></Input>
				</div>

				<Button type="submit">Register</Button>
      </form>
			</>
		)
	}
}