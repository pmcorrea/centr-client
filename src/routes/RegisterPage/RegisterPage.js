import React, {Component} from 'react'
import { Button, Input } from "../../Utils/Utils";
import './RegisterPage.css'
import MainContext from "../../contexts/MainContext";
import AuthApiService from "../../services/auth-api-service";
import BG from './login_background.png'


export default class RegisterPage extends Component {

	static contextType = MainContext

	state = { 
		error: null,
		user_name: null,
		password: '',
		passwordConfirm: '',
		visibility: 'Private',
		showPasswordValidationBox: true,
		username_exist: true, 
		profilePicURL: ''
	};

	onRegisterSuccess = () => {
		this.props.history.push("/");
	}

	handleRegister(e) {
		e.preventDefault()

		this.setState({ error: null })
		const {user_name, password, visibility, profilePicURL} = this.state

		AuthApiService.postRegister({
			user_name: user_name, 
			password: password,
			visibility: visibility,
			profilePicURL: profilePicURL
		})
		.then(res => {
			this.onRegisterSuccess();
		})
		.catch(res => {
			this.setState({ error: res.error });
		})
	}

	setUsername(username) {
		this.setState({
			user_name: username
		}, () => this.checkUsername())
	}

	checkUsername() {
		const {user_name} = this.state

		if(user_name.length > 4) {
			AuthApiService.checkUsername({
				user_name: user_name
			})
			.then(res => {
				this.setState({
					username_exist: res
				})
			})
			.catch(res => {
				this.setState({ error: res.message });
			})
		} 

		else {
			this.setState({
				username_exist: true
			})
		}		
	}

	setPassword(password) {
		this.setState({
			password: password
		}, () => {
			this.validatePassword()
			this.matchPasswords()
		})
	}

	setPasswordConfirm(password) {
		this.setState({
			passwordConfirm: password
		}, () => {
			this.validatePassword()
			this.matchPasswords()
		})
	}

	validatePassword() {
		let pw = this.state.password
		// let pwPattern = /(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&])[\S]+/
		let pwPattern = /(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&])[\S]+/
		let whiteSpacePattern = /(\s)+/
		let length = pw.length >= 8
		let result = pwPattern.test(pw)
		let result2 = whiteSpacePattern.test(pw)

		if (result && !result2 && length) {
			this.setState({
				showPasswordValidationBox: false
			})
		} else {
			this.setState({
				showPasswordValidationBox: true
			})
		}
	}

	matchPasswords() {
		let pw = this.state.password
		let pwc = this.state.passwordConfirm

		let match = pw === pwc
		return match
	}

	disableSubmitButton() {
		if (!this.state.showPasswordValidationBox && !this.state.username_exist) {
			let match = this.matchPasswords()

			if (match) {
				return false
			}
			return true
			
		} return true
	}



	render() {

		


		const { error } = this.state;
		const { showPasswordValidationBox } = this.state;
		return (
			<>
			<div className="registration__container">

				<h1>REGISTER</h1>
				<img src={BG} className="bg_image" alt="background"/>
				<div className="RegisterForm__form_contianer">
				
				<form className="RegisterForm" onSubmit={e => this.handleRegister(e)} id="form_register">
	
				<div role="alert">{error && <p className="red">{error}</p>}</div>

				<div className="user_name">
					{/* <label htmlFor="LoginForm__user_name">User name</label> */}
					<Input 
					className='input_field'
					required 
					placeholder="username"
					style={{
						borderColor: this.state.username_exist ? 'red' : 'green',
						textDecoration: this.state.username_exist ? 'line-through' : 'none'
					}}
					name="user_name" id="LoginForm__user_name" onChange={e => this.setUsername(e.target.value)}></Input>
				</div>

				<div className="password">
				{/* <label htmlFor="LoginForm__password">Password</label> */}
				<Input
					className='input_field'
					required
					name="password"
					type="password"
					placeholder="password"
					id="LoginForm__password"
					style={{borderColor: this.disableSubmitButton() ? 'red' : 'green'}}
					onChange={
						e => {
							this.setPassword(e.target.value)
						}}
						></Input>
				<Input
				className='input_field'
				required
				name="password_confirm"
				type="password"
				placeholder="password"
				id="LoginForm__password_confirm"
				style={{borderColor: this.disableSubmitButton() ? 'red' : 'green'}}
				onChange={
					e => 
					this.setPasswordConfirm(e.target.value)
				}
				></Input>
				</div>

				{/* <label id="file-upload_label">Profile pic</label>
				<Input id="file-upload" type="file" onChange={(e) => this.setProfilePic(e.target)}></Input> */}

				

				<Button className='register_button' type="submit" disabled={
					this.disableSubmitButton()}
					>Register</Button>

{showPasswordValidationBox && 
				<div role="alert" className="registration__validation_box">
					<p className="validation_rules">Password must include:</p>
					<p className="validation_rules">8 characters minimum, no spaces</p>
					<p className="validation_rules">Uppercase and lowercase letters</p>
					<p className="validation_rules">Special symbol and numbers</p>
				</div>
				}

      </form>
				</div>
	  </div>
			</>
		)
	}
}