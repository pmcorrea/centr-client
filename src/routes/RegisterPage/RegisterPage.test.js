import RegisterPage from './RegisterPage.js'
import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('App test suite', () => {
	// Smoke Test
	it('renders without crashing', () => {
		const div = document.createElement('div')

		ReactDOM.render(
			<RegisterPage />,
			div
		)
		ReactDOM.unmountComponentAtNode(div)
	})

	// Snapshot Test
	it('renders the snapshot', () => {
		const wrapper = shallow(<RegisterPage />)
		expect(toJson(wrapper)).toMatchSnapshot()
	})
})