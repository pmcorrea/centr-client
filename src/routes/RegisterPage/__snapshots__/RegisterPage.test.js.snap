// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test suite renders the snapshot 1`] = `
<Fragment>
  <div
    className="registration__container"
  >
    <h1>
      REGISTER
    </h1>
    <img
      alt="background"
      className="bg_image"
      src="login_background.png"
    />
    <div
      className="RegisterForm__form_contianer"
    >
      <form
        className="RegisterForm"
        id="form_register"
        onSubmit={[Function]}
      >
        <div
          role="alert"
        />
        <input
          className="input_field"
          id="LoginForm__user_name"
          name="user_name"
          onChange={[Function]}
          placeholder="username"
          required={true}
          style={
            Object {
              "borderColor": "red",
              "textDecoration": "line-through",
            }
          }
        />
        <input
          className="input_field"
          id="LoginForm__password"
          name="password"
          onChange={[Function]}
          placeholder="password"
          required={true}
          style={
            Object {
              "borderColor": "red",
            }
          }
          type="password"
        />
        <input
          className="input_field"
          id="LoginForm__password_confirm"
          name="password_confirm"
          onChange={[Function]}
          placeholder="password"
          required={true}
          style={
            Object {
              "borderColor": "red",
            }
          }
          type="password"
        />
        <button
          className="register_button"
          disabled={true}
          type="submit"
        >
          Register
        </button>
        <div
          className="registration__validation_box"
          role="alert"
        >
          <p
            className="validation_rules"
          >
            Password must include:
          </p>
          <p
            className="validation_rules"
          >
            8 characters minimum, no spaces
          </p>
          <p
            className="validation_rules"
          >
            Uppercase and lowercase letters
          </p>
          <p
            className="validation_rules"
          >
            Special symbol and numbers
          </p>
        </div>
      </form>
    </div>
  </div>
</Fragment>
`;
